### ISSUE TITLE
Markers getting duplicated

### ISSUE THREAD

**Issue by IvanCoronado** [angular-ui/angular-google-maps#1574]
I am working on  [this project](http://htmlpreview.github.io/?https://rawgit.com/IvanCoronado/running-statistics-from-Barcelona/master/dist/index.html#/).

I am overwritting defaults markers with labelClass. CSS is working fine but i am getting markers duplicated.
![image](https://cloud.githubusercontent.com/assets/13351933/10613157/52ca9036-7754-11e5-8245-1715e6134dbe.png)

I have just 6 markers but if u check the console i am getting 12 markers, also one of them is getting 0.1 opacity without any reason.

The code is [here](https://github.com/IvanCoronado/running-statistics-from-Barcelona) and [this link](http://stackoverflow.com/questions/22202569/google-map-api-multiple-icons-in-wrong-spot) could be usefull.

**Comment by nmccready** [comment]
Actually you could probably easily write a spec against this in one of the markers specs doing the same thing you just did. I'll try to copy your controller and dig into this eventually. I will try to look at it this weekend.

**Comment by nmccready** [comment]
Yeah as you state this appears to be a bug with the labelMarkers plugin.

**Comment by nmccready** [comment]
BTW where is mapStyles.shades coming from? What module provides the service?

**Comment by nmccready** [comment]
So anyway if you can find a better markerLabel library we can point curl.json or bower.json to that. Also we can modify MarkerLabel to not be global if needed and pulled in as a service.

But I myself do not have plans on fixing this.

**Comment by nmccready** [comment]
If you plan on doing work against this I can put it in the 2.2.3 milestone; or now (2.2.2) if you can do it in a few days .

**Comment by nmccready** [comment]
My suggestion is to copy this [project]( https://github.com/googlemaps/js-map-label) as a template and port markerWithLabel to that. Then you can debug and have your on repo. I do not see where markerWithLabel has been ported yet.

**Comment by nmccready** [comment]
Here is one that might be worth it; https://github.com/jesstelford/node-MarkerWithLabel/commits/master looks like they might have fixed the issue too.

**Comment by IvanCoronado** [comment]
What MarkerWithLabel version are you using?

**Comment by nmccready** [comment]
So if that new fork be @jesstelford is on bower or npm then we should reference that.

**Comment by IvanCoronado** [comment]
I hardcoded this commits https://github.com/jesstelford/node-MarkerWithLabel/commit/ca93914177ca644ba868d911e29539d9978b6a73 https://github.com/jesstelford/node-MarkerWithLabel/commit/5c355817a0bbd4c65dd7363e3bb328a9ed085803 on my bower components version of your library and i still have the bug.
I even tried with http://google-maps-utility-library-v3.googlecode.com/svn/tags/markerwithlabel/1.1.9/markerwithlabel/src/markerwithlabel.js and no way.

**Comment by nmccready** [comment]
Are you sure grunt is using it? The whole wrapping / injecting MarkeWithLabel expects that lib to be in certain places via grunt. So since your using a diff library you must update it.

**Comment by nmccready** [comment]
- Where curl components hey [concated to](https://github.com/angular-ui/angular-google-maps/blob/master/grunt/options.coffee#L182-L183)

- then [injected](https://github.com/angular-ui/angular-google-maps/blob/master/grunt/options.coffee#L266-L272)

Your going to need to understand the grunt if your going to fix this.

**Comment by IvanCoronado** [comment]
But you finally get this https://github.com/angular-ui/angular-google-maps/blob/master/dist/angular-google-maps.js
My local is using that file, i can edit it. Am i wrong?

**Comment by IvanCoronado** [comment]
The bug is here:
![captura_de_pantalla_102215_062230_pm](https://cloud.githubusercontent.com/assets/13351933/10671545/65507e58-78eb-11e5-9fe4-5d4104bb7321.jpg)

I am getting a dom element attached to the icon marker and to the cross marker.

**Comment by nmccready** [comment]
Do you have a fork where you can show me lines of code?

**Comment by IvanCoronado** [comment]
Ok i found the problem, i will try to explain it in english :S

The problem is on the MarkerWithLabel library but really isn't a bug, its just an impossibility (with this library).
If you check the library we see this comment
![captura_de_pantalla_102215_073436_pm](https://cloud.githubusercontent.com/assets/13351933/10673026/f03739b4-78f3-11e5-8ebc-480c60247068.jpg)
They are creating a div for the label and an "invisible" div (*) for handling mouse events...

*It isn't invisible
![captura_de_pantalla_102215_075013_pm](https://cloud.githubusercontent.com/assets/13351933/10673440/1ebe9096-78f6-11e5-9212-76e590fdd7df.jpg)

In conclusion, i am doing hover about this invisible div which has 0.1 opacity.

I just modified the library so i don't create invisible div anymore and also i intercept events with real label. Now is working.

**Comment by nmccready** [comment]
So should we use another fork of the library or not? Posting github line number is way more helpful then coping and pasting images btw.

**Comment by IvanCoronado** [comment]
You must choose between to hover labels by css or to capture events even if the label is in the shadow of a google.maps.InfoWindow.

Sorry i am a github noob, i will check how i can post line number.

**Comment by nmccready** [comment]
@IvanCoronado like this https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/marker.coffee#L15

any file after #L15, means line 15 . Also when u look at the file just click to high light the lines. Hold shift and click to highlight many. Its all auto.

**Comment by IvanCoronado** [comment]
Ok if someone wants to style marker label with css hover just need change [this lines](https://github.com/angular-ui/angular-google-maps/blob/master/dist/angular-google-maps.js#L11335-L11917) on his .\bower_components\angular-google-maps\dist\angular-google-maps.js  with this [code](https://github.com/IvanCoronado/MarkerWithLabelCSSHoverable).

**Comment by IvanCoronado** [comment]
Thanks you for all @nmccready :)
You are doing amazing work with angular-google-maps directive.
If you check my [project](http://htmlpreview.github.io/?https://rawgit.com/IvanCoronado/running-statistics-from-Barcelona/master/dist/index.html#/) now, the hover is working.

**Comment by nmccready** [comment]
Got you, like I said that is injected which I can fix. However, can your fork be created from jesstelford's as he has more fixes. Or better yet PR your fix to his repo. I would prefer you at least have his changes. Anyway it needs to be published to bower, npm, and tagged on git.

**Comment by nmccready** [comment]
Actually, I'll just fork his to my own repo. Then I'll merge your changes into his. Thus the fork being mine.

**Comment by IvanCoronado** [comment]
Ok, i can do it too if you want. (And you help me xD)

**Comment by jesstelford** [comment]
Ahoy!

@nmccready Feel free to open a PR against jesstelford/node-MarkerWithLabel if you like. If it's good quality, I'm also happy to give you commit rights.

Since I also have the [npm package](https://www.npmjs.com/package/markerwithlabel), I can then publish any new versions with bug fixes / enhancements :)

**Comment by nmccready** [comment]
Sweet thanks @jesstelford

**Comment by nmccready** [comment]
@IvanCoronado make the PR so you can get credit for the change. If I don't here back from you by COB ~1600 EST tomorrow I will do it over the weekend.

**Comment by nmccready** [comment]
@IvanCoronado it would be a lot easier to see your code differences if your initial commit to your project did not include your modifications. Now I must look at the code line by line to find the changes. Which honestly I do not have the time for. Please fork @jesstelford project and then make the changes. Then PR to his fork. I am closing this until I get some feedback.

**Comment by IvanCoronado** [comment]
I did that https://github.com/jesstelford/node-MarkerWithLabel/pull/5
I am waiting @jesstelford answer

**Comment by nmccready** [comment]
Apologies, got lost and frustrated in the feed; plus a screaming kid in the
background ;)

**Comment by IvanCoronado** [comment]
hehe no problem, i will notify you anyway.

**Comment by IvanCoronado** [comment]
Google library accepted my PR, https://github.com/googlemaps/v3-utility-library/tree/master/markerwithlabel/src

**Comment by nmccready** [comment]
We should probabaly go with google's @jesstelford can you PR your updates to google as well?

**Comment by artuska** [comment]
I have the same issue — markers duplicates on map zoom — it happens when you set `labelContent` with the dynamically created element, e.g.:

```
var myDiv = document.createElement('div');
    myDiv.innerHTML = 'foobarbaz';

var markerConfig = {
        map: this.map,
        clickable: true,
        draggable: false,
        position: new google.maps.LatLng(this.coords.lat, this.coords.lng),
        labelAnchor: new google.maps.Point(0, 30),
        labelClass: 'gm-offer-marker',
        labelContent: myDiv,
        icon: {
            anchor: new google.maps.Point(0, 0),
            path: google.maps.SymbolPath.CIRCLE,
            scale: 0,
            fillColor: '#000000',
            fillOpacity: 1,
            strokeColor: '#000000',
            strokeWeight: 0
        }
    };

this.marker = new MarkerWithLabel(markerConfig);
```

If I set `labelContent` to a string or a number or an inline HTML (e.g. `labelContent: '<div class="myClass">foobarbaz</div>'`) everything works perfectly, nothing duplicates on zoom.

**Comment by artuska** [comment]
I have updated Angular Google Maps to the latest 2.3.2 version, but MarkerWithLabel (which is included in Angular Google Maps by default) is still stuck at version 1.1.10, not the latest 1.1.13.

**Comment by nmccready** [comment]
@artuska feel free to use your computer skills to submit a PR and update the library.
